flink-connector-gcp-pubsub/src/main/java/org/apache/flink/streaming/connectors/gcp/pubsub/PubSubSink.java

```diff
@@ -62,6 +62,7 @@
  *
  * @param <IN> type of PubSubSink messages to write
  */
+@Deprecated
 public class PubSubSink<IN> extends RichSinkFunction<IN> implements CheckpointedFunction {
```

Comment:
👍

Reply:
When is good to remove these API?

Reply:
We can remove this when we release a new major version `4.0` once this is merged we cann follow up with another Jira to remove